### ISSUE TITLE
Serial enumeration

### ISSUE THREAD

**Issue by Susurrus** [dcuddeback/serial-rs#13]
Looks to be [some registry parsing for Windows](http://code.qt.io/cgit/qt/qtserialport.git/tree/src/serialport/qserialportinfo_win.cpp#n63), but for Unix it's merely [processing /dev/ entries](http://code.qt.io/cgit/qt/qtserialport.git/tree/src/serialport/qserialportinfo_unix.cpp#n59).

**Comment by goertzenator** [comment]
Python's take on [serial port enumeration](http://pythonhosted.org/pyserial/tools.html).

**Comment by dcuddeback** [comment]
This is definitely in the plans. I've done some groundwork for this already by implementing crates needed for listing serial ports on Linux (libudev) and OS X (IOKit and CoreFoundation). Using those APIs seems similar to Python's solution, which mentions using sysfs (libudev) and IOKit. I pasted some links in #14 in case someone wants to start working on this before me.

**Comment by Susurrus** [comment]
@dcuddeback Can you respond to my PR so we can ideally merge this code? I've been using it for a while in my own project and it works on Windows and Linux (appveyor fails because of another package not working on Rust 1.3).

**Comment by dcuddeback** [comment]
@Susurrus I feel like my input has been mostly ignored up to this point, so I think it's a better use of my free time to work on this feature myself than to spend more time trying to guide the existing PRs into an acceptable solution.

**Comment by Susurrus** [comment]
Fair enough. In the mean time I'll use my fork for this functionality then. Any idea when this might be available in mainline? Additionally, let me know if there's any way I can help to speed this along beyond what I've already done.

**Comment by Susurrus** [comment]
@dcuddeback Any updates on this work and the big refactoring? I'm looking to push my project to crates.io, which requires that it only uses libraries with releases on there, so I'm blocking on serial enumeration in a release.

**Comment by danieledapo** [comment]
@dcuddeback any news about this? It would be a great addition to the library.

**Comment by Michael-F-Bryan** [comment]
@dcuddeback is anyone working on this at the moment? If not, I might be able to look into it and make a PR.

**Comment by AJAnderson** [comment]
Bump again for work? This is pretty important functionality for finding devices in a robust manner.
